tests/integration_tests/tests/agentless_tests/test_usage_collector.py

```diff
@@ -0,0 +1,116 @@
+########
+# Copyright (c) 2013-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from os.path import join
+
+from integration_tests import AgentlessTestCase, AgentTestCase, BaseTestCase
+from integration_tests.framework import docl
+from integration_tests.tests.utils import get_resource as resource
+
+
+MANAGER_PYTHON = '/opt/manager/env/bin/python'
+COLLECTOR_SCRIPTS = ['collect_cloudify_uptime', 'collect_cloudify_usage']
+SCRIPTS_DESTINATION_PATH = '/opt/cloudify/usage_collector'
+LOG_PATH = '/var/log/cloudify/usage_collector'
+LOG_FILE = 'usage_collector.log'
+
+
+class TestUsageCollector(AgentlessTestCase):
+
+    def test_collector_scripts(self):
+        messages = [
+            "Uptime script finished running",
+            "Usage script finished running",
+            "'customer_id': 'mock_customer'",
+            "'node_instances_count': 1L",
+            "'compute_count': 1L",
+            "'agents_count': 0L",
+            "'premium_edition': True"
+        ]
+        self._test_base("dsl/basic.yaml", messages)
+
+    def test_multi_instance(self):
+        messages = [
+            "'node_instances_count': 4L",
+            "'compute_count': 2L",
+        ]
+        self._test_base("dsl/multi_instance.yaml", messages)
+
+    def test_scaled(self):
+        messages = [
+            "'node_instances_count': 6L",
+            "'compute_count': 2L",
+        ]
+        self._test_base("dsl/scale5.yaml", messages)
+
+    def test_compute_not_started(self):
+        deployment = BaseTestCase.deploy(resource("dsl/multi_instance.yaml"))
+        docl.execute('mkdir -p {0}'.format(LOG_PATH))
+        docl.execute('echo > {0}'.format(join(LOG_PATH, LOG_FILE)))
+        for script in COLLECTOR_SCRIPTS:
```

Comment:
All the mutual code with `_test_base ` can be in a helper function like `_run_collector_scripts`